### PR TITLE
Move support for sparse fields into a separate module

### PR DESCRIPTION
**Pull request by rco-odoo**
This removes the support for sparse and serialized fields into module `sparse_fields`. The process has led to an interesting refactoring of the model and fields reflection into `ir.model` and `ir.model.fields`.

This provides an alternative to #14446.

### CODE REVIEW

**Comment by bealdav** [comment]
Nice by implicit naming module convention is no plural, no ? sale, purchase, etc (few exceptions)
Please rename to sparse_field, thanks

**Comment by rco-odoo** [comment]
@bealdav most modules are named after business domains. This one is named after the feature it provides. I think that we should simply choose a correct English term or expression that makes sense. Note by the way that some existing module names suck (`sale` should be `sales`, `account` should be `accounting`).

Note that I prefixed it with `base_` to suggest that it is a "frameworky" module.

**Comment by bealdav** [comment]
@rco-odoo  I understand your point of view as editor voice.

But use plural for module names is prone error for developers in community 
look there https://github.com/odoomrp/odoomrp-wip/issues/431#issuecomment-68916386 and also anywhere in odoo ecosystem
OCA try to follow Odoo editor's rules, even implicit ones like module names. It's not so easy, to have uniformity on the whole ecosystem. Uniformity is easy to remember, not specific cases.

Odoo SA have short module names, not community `sale_order_recalculate_whatever_prices_anything`.

Define which word must be plural (without any error) is in practice not easy for developers for some case. Community (@pedrobaeza) would like to avoid waste it's time in spelling where the plural should be.

You can have folder name as `sale` and manifest name key as `Sales` and all is ok

Then when you have error on not mandatory details, the easier is to remove these details, isn't it ? We have some real problem elsewhere, I think

Yes Odoo has a community, I hope it'll be taken in account.

Thanks

**Comment by rco-odoo** [comment]
@bealdav got it.

**Comment by wtaferner** [comment]
I agree with @bealdav that the technical name should somehow follow a convention easy to remember and  to follow...even Odoo does this in its model naming. Singular does help to skip complexity. Anyway, the ```base_``` is also a good additional information that it is about an extension of the framework itself.

**Comment by pedrobaeza** [comment]
Raphael, very kind by your part. Thank you!

**Comment by bealdav** [comment]
Yes Raphael, really very kind by your part.
 
Thanks a lot from community
 
Spread the word ;-)
 
Not a lot modules to convert in master :1st_place_medal: : 
- website_sale_options, 
- website_slides, 
- stock_landed_costs
- website_event_questions
- sale_order_dates
- pos_data_drinks

Only some recent modules

File: addons/base_sparse_field/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
+
+from . import fields
+from . import models
```
Comment (Yajo) [inline addons/base_sparse_field/models/__init__.py]: Did you realize that due to #8591 monkey patching in fields file is going to happen no matter the add-on is installed or not? Then why split it?

**Pull request merged**